kolibri/core/assets/src/components.js

```diff
@@ -149,6 +73,132 @@ var CrudCollection = Mn.CollectionView.extend({
 });
 
 
+var CrudAddItem = Mn.ItemView.extend({
+    // Template should either be the html *or* (as it is here) be a *function* that returns the rendered html.
+    // If it's a function, it's passed an object which is the "serialized" model of the view instance.
+    // In other words, if the view instance is given a model attribute when instantiated, then the argument
+    // to this function will be a copy of that model's .attributes hash.
+    template: function() {
+        var html =
+            '{{#unless valid}}<span>Invalid Input!</span><br />{{/unless}}' +
+            '{{#each create}}' +
+                '<input data-attr="{{ name }}" type="text" placeholder="{{ name }}" value="{{ default }}" /><br />' +
+            '{{/each}}' +
+            '<button class="create flat-button">Create</button>';
+        return Handlebars.compile(html)({create: this.create, valid: this.valid});
```

Comment:
'to create' might even be implied .. it might simply be called `this.item_fields` or something.

Of course none of this is relevant to Mn in particular. However careful naming can help frame views declaratively vs imperatively. In this case the emphasis would be on _what_ (user-editable item attributes) versus _how_ (creation of DOM elements).